.github/workflows/merge-commits.yml

```diff
@@ -72,13 +72,37 @@ jobs:
           git fetch --all
           git reset --hard ${{needs.dequeue.outputs.SHA}}
           git push --set-upstream origin main-next-${{needs.dequeue.outputs.SHA}}
+  merge-next:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      packages: write
+      actions: write
+      checks: write
+      issues: write
+      pull-requests: write
+      statuses: write
+    needs: [ create-branch ]
+    steps:
+      - name: Git checkout
+        uses: actions/checkout@7884fcad6b5d53d10323aee724dc68d8b9096a2e # pin@v2
+        with:
+          fetch-depth: '0'
+      - name: Merge next
+        run: |
+          git config user.name $USERNAME
+          git config user.email $EMAIL
+          git fetch origin
+          git checkout main-next-${{needs.dequeue.outputs.SHA}}
+          git merge $TARGET_BRANCH -X --no-ff
```

Comment:
So if this merge fails, how does it fall back to opening the PR with the old HEAD? Is that the `always()` condition below? It will always run that job even if merge-next fails, and the HEAD will be on the main-branch commit at that point? I'd add comments explaining that.

Reply:
that's a good point. I have first added the step "merge-dry-run" to check if git merge succeeds and not pushing the merge to the branch. only if it succeeds, it's pushed to our main-next branch otherwise the PR should be pointing to the old HEAD before the merge was made right?